File: 0430/캐시.py
from collections import deque

def solution(cacheSize, cities):
    answer = 0
    cach = deque()
    if cacheSize != 0:
        for city in cities:
            city = city.lower() # 대소문자 구분 없게 하려면 이게 필수!! or upper를 사용하던가~
            if len(cach) < cacheSize: # 일단 더 넣을 수 있다면
                if city in cach:
                    answer += 1
                else:
                    answer += 5
                # 캐시에 있다면 1플러스, 없다면 5플러스
                cach.append(city)
            else: # 캐시가 일단 꽉 차있다면!!!
                if city in cach:
                    answer += 1
                    cach.remove(city)
                    cach.append(city)
                    # 이거 솔직히 될지 잘 모르겠어...
                else:
                    # 캐시에 안 들어있으면!!!
                    answer += 5
                    cach.popleft() # 왼쪽에서 하나 뺴야징
                    cach.append(city) # 새로 들어온거 
                # 똑같은 로직입니다.
    else:
        answer = len(cities) * 5

    return answer

File: 0430/캐시.js
function solution(cacheSize, cities) {
    var answer = 0;
    const cach = [];
    if (cacheSize !== 0) {
        cities.forEach(city => {
            city = city.toLowerCase()
            // console.log(city)
            if (cach.length < cacheSize){
                if (city in cach) {
                    answer += 1
                } else {
                    answer += 5
                }
                cach.push(city)
            } else {
                if (city in cach) {
                    answer += 1
                    // cach.remove(city)   // remove가 되나? 안되는거같어 실패작.
                    const idx = cach.indexOf(city)
                    cach.splice(idx, 1) // 이렇게 하라는 거지?
                    cach.push(city)
                } else {
                    answer += 5
                    cach.shift()
                    // cach.shift() 같은거?
                    cach.push(city)
                }
            }
        });
    } else {
        answer = cities.length * 5
    }
    console.log(cach)
    return answer;
}